TensorMath.lua

```diff
@@ -51,6 +51,18 @@ static int torch_isnonemptytable(lua_State *L, int idx)
 }
 ]])
 
+
+interface:print([[
+static const void* torch_istensorarray(lua_State *L, int idx)
+{
+  if (!torch_isnonemptytable(L, idx)) return 0;
+
+  lua_checkstack(L, 1);
+  lua_rawgeti(L, 1, 1);
+  return (torch_istensortype(L, luaT_typename(L, -1)));
```

Comment:
Shouldn't you call `lua_remove` here instead of below? If `torch_istensortype` returns NULL, the `lua_remove` below isn't getting executed, so you leave the table element on the stack.